fili-core/src/test/groovy/com/yahoo/bard/webservice/table/availability/MetricUnionAvailabilitySpec.groovy

```diff
@@ -0,0 +1,189 @@
+// Copyright 2017 Yahoo Inc.
+// Licensed under the terms of the Apache license. Please see LICENSE.md file distributed with this work for terms.
+package com.yahoo.bard.webservice.table.availability
+
+import com.yahoo.bard.webservice.data.config.names.TableName
+
+import com.google.common.collect.Sets
+import com.yahoo.bard.webservice.data.metric.MetricColumn
+import com.yahoo.bard.webservice.table.PhysicalTable
+import com.yahoo.bard.webservice.table.PhysicalTableSchema
+import com.yahoo.bard.webservice.table.resolver.DataSourceConstraint
+import com.yahoo.bard.webservice.util.SimplifiedIntervalList
+
+import org.joda.time.Interval
+
+import spock.lang.Specification
+import spock.lang.Unroll
+
+/**
+ * Test for metric union availability behavior.
+ */
+class MetricUnionAvailabilitySpec extends Specification {
+    Availability availability1
+    Availability availability2
+
+    MetricColumn metricColumn1
+    MetricColumn metricColumn2
+
+    Interval interval1
+    Interval interval2
+    Interval interval3
+    Interval interval4
+
+    PhysicalTableSchema schema1
+    PhysicalTableSchema schema2
+
+    PhysicalTable physicalTable1
+    PhysicalTable physicalTable2
+
+    MetricUnionAvailability metricUnionAvailability
+
+    def setup() {
+        availability1 = Mock(Availability)
+        availability2 = Mock(Availability)
+
+        metricColumn1 = new MetricColumn('metric1')
+        metricColumn2 = new MetricColumn('metric2')
+
+        interval1 = new Interval('2018-01-01/2018-02-01')
+        interval2 = new Interval('2018-02-01/2018-03-01')
+        interval3 = new Interval('2019-01-01/2019-02-01')
+        interval4 = new Interval('2018-11-01/2018-12-01')
+
```

Comment:
Sort of.  They aren't ordered.  And they don't overlap at all.  So they feel kind of arbitrary.  Is there some meaning that they have relative to each other?